storey/flow.py

```diff
@@ -608,73 +611,118 @@ async def _handle_completed(self, event, response):
 
 
 class _Batching(Flow):
-    def __init__(self, max_events: Optional[int] = None, timeout_secs=None, **kwargs):
+    def __init__(
+            self,
+            max_events: Optional[int] = None,
+            timeout_secs: Optional[int] = None,
+            group_by_key: bool = False,
+            key: Optional[Union[str, callable]] = None,
+            **kwargs,
+    ):
         super().__init__(**kwargs)
 
-        self._max_events = max_events
-        self._event_count = 0
-        self._batch = []
-        self._batch_time = None
-        self._timeout_task = None
-
+        self._max_events: int = max_events
         self._timeout_secs = timeout_secs
+        self._group_by_key = group_by_key
+
         if self._timeout_secs is not None and self._timeout_secs <= 0:
-            raise ValueError('Batch timeout cannot be 0 or negative')
+            raise ValueError("Batch timeout cannot be 0 or negative")
+
+        self._event_count: Dict[Optional[str], int] = defaultdict(int)
+        self._batch: Dict[Optional[str], List[Any]] = defaultdict(list)
+        self._batch_time: Dict[Optional[str], datetime] = {}
```

Comment:
Does it make sense for the key to be `None`?

Reply:
In case it's not grouped by key